### PR TITLE
MVC Tasks Dedenok

### PR DESCRIPTION
**Pull request by viktoriiadedenok**

### CODE REVIEW

File: FullName/FullName/Controllers/NamesController.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        {
+            name.Id = GenereteId();
+            names.Add(name);
+            return RedirectToAction("ShowName");
+        }
+
+        [HttpPost]
+        public ActionResult Delete(int id)
+        {
+            names.Remove(names.FirstOrDefault(n => n.Id == id)) ;
```
Comment (gandzha) [inline FullName/FullName/Controllers/NamesController.cs:41]: The case when name could not be found should be handled correctly.

File: FullName/FullName/HttpHandler/ImageHandlerMiddleWare.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+            var path = context.Request.Path.ToString();
+          
+            var responce = context.Response.Body;
+           
+            string[] formats = new string[] { ".ai", ".bmp", ".gif", ".ico", ".jpeg", ".jpg", ".png", ".ps", ".psd", ".svg", ".tif", ".tiff", ".jfif" };
+          
+            var resPath = @$"{Directory.GetCurrentDirectory()}\Images\not_allowed.png";
+     
+            var photo = File.ReadAllBytes(resPath);
```
Comment (gandzha) [inline FullName/FullName/HttpHandler/ImageHandlerMiddleWare.cs:32]: Photo is read even if a request is not directed to an image file. For optimization purpose this code should be moved under _if(res)_ condition

File: FullName/FullName/Infrastructure/CustomNameModelBinderProvider.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FullName.Infrastructure
+{
+    public class CustomNameModelBinderProvider: IModelBinderProvider
+    {
+        public IModelBinder GetBinder(ModelBinderProviderContext context)
+        {
+            // Для объекта SimpleTypeModelBinder необходим сервис ILoggerFactory
```
Comment (gandzha) [inline FullName/FullName/Infrastructure/CustomNameModelBinderProvider.cs:17]: At least, you could have cleaned these comments from copy/paste code.

File: FullName/FullName/Views/BindeNames/ShowBindedNames.cshtml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+@foreach (var item in Model) {
+        <tr>
+            <td>
+                @Html.DisplayFor(modelItem => item.BindedFirstName)
+            </td>
+            <td>
+                @Html.DisplayFor(modelItem => item.BindedLastName)
+            </td>
+            <td>
+                @Html.ActionLink("Edit", "Edit", new { /* id=item.PrimaryKey */ }) |
```
Comment (gandzha) [inline FullName/FullName/Views/BindeNames/ShowBindedNames.cshtml:31]: These actions are not related to any item. What will they do?

File: FullName/FullName/Views/Names/AjaxSearch.cshtml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    $(function () {
+        $("#btn-search").click(function () {
+           $("#tblNames tbody").remove();
+            var std = {};
+            std.FirstName = $("#SearchWord").val();
+            $.ajax({
+                type: "POST",
+                url: '@Url.Action("FindSearched")',
+                data: '{"FirstName":"' + std.FirstName + '"}',
```
Comment (gandzha) [inline FullName/FullName/Views/Names/AjaxSearch.cshtml:46]: It looks like not working code as you send FirstName(?) here and expect Name on the backend.

File: FullName/FullName/Views/Value/ShowNumber.cshtml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+﻿
+@{
+    ViewData["Title"] = "ShowNumber";
+}
+
+<h1>123</h1>
```
Comment (gandzha) [inline FullName/FullName/Views/Value/ShowNumber.cshtml:6]: Why hardcode?

File: FullName/FullName/Views/Names/AjaxSearch.cshtml (hunk 5 above)
Comment (viktoriiadedenok) [reply FullName/FullName/Views/Names/AjaxSearch.cshtml:46]: It should work by pushing the "search" button, and does not work pressing "enter" button.

**Comment by viktoriiadedenok** [comment]
Dedenok task fixed
